src/Model/Resolver/Products/DataPostProcessor/Attributes.php

```diff
@@ -209,41 +207,44 @@ protected function appendWithOptions(
 
         // To collect ids of options, to later load swatch data
         $optionIds = [];
+        $formattedAttributes = [];
+        $swatchAttributes = [];
 
-        // Loop again, get options sorted in the right places
-        /** @var Product $product */
-        foreach ($products as $product) {
-            $id = $product->getId();
+        // Format attribute to use them later
+        foreach($detailedAttributes as $attribute) {
+            $formattedAttributes[] = [
+                'attribute_id' => $attribute->getAttributeId(),
+                'attribute_code' => $attribute->getAttributeCode(),
+                'attribute_options' => $attribute->getOptions()
```

Comment:
I assume not all attributes will even have options in the first place. From what I remember, the attribute list provided/loaded in $detailedAttributes wasn't limited to the ones with options.
No options, can skip attribute entirely in this method.